test/functional/Java8andUp/src_80/org/openj9/test/java/lang/Test_Class.java

```diff
@@ -377,6 +377,70 @@ public Object run() {
 		java.security.AccessController.doPrivileged(action);
 	}
 
+	static class ParentClass {
+		public static void methodPublicStatic() {}
+		public void methodPublicInstance() {}
+		static void methodPkgPrivateStatic() {}
+		void methodPkgPrivateInstance() {}
+		private static void methodPrivateStatic() {}
+		private void methodPrivateInstance() {}
+	}
+
+	static class ChildClass extends ParentClass {
+		public static void methodPublicStatic() {}
+		public void methodPublicInstance() {}
+		static void methodPkgPrivateStatic() {}
+		void methodPkgPrivateInstance() {}
+		private static void methodPrivateStatic() {}
+		private void methodPrivateInstance() {}
+	}
+
+	static interface SuperInterface {
+		void methodPublicInterface(); 
+		public static void methodPublicStatic() {};
+	}
+
+	static interface SubInterface extends SuperInterface {
+		void methodPublicInterface(); 
+		public static void methodPublicStatic() {};
+	}
+
+	static class ClassImplInterface implements SubInterface {
+		public void methodPublicInterface() {}
+		public static void methodPublicStatic() {};
+	}
+	
+	@Test
+	public void test_getMethods_subtest3() {
+		Method[] methods = ChildClass.class.getMethods();
+		for (Method method : methods) {
+			String declaringClassName = method.getDeclaringClass().getName();
+			if (declaringClassName.indexOf("ParentClass") != -1) {
+				Assert.fail("Should not return Method " + method.getName() + " declared in class " + declaringClassName);
+			}
+		}
+	}
+	@Test
+	public void test_getMethods_subtest4() {
+		final Method[] methods = SubInterface.class.getMethods();
+		for (Method method : methods) {
+			String declaringClassName = method.getDeclaringClass().getName();
```

Comment:
Why not compare to the actual java.lang.Class rather than its name?  ie:
`if (method.getDeclaringClass().equals(SuperInterface.class)) { ...`

Reply:
Agreed, comparing with `j.l.Class`.